File: android/app/src/main/kotlin/com/callblocker/model/PatternType.kt
package com.callblocker.model

enum class PatternType {
    /** Exact 10-digit number, e.g. "0912345678" */
    EXACT,

    /** Leading digits that must match, e.g. "0900" blocks all 0900xxxxxx */
    PREFIX,

    /** Fixed-length pattern where 'x' matches any single digit, e.g. "09xx1234" */
    WILDCARD;

    val displayName: String get() = when (this) {
        EXACT    -> "Exact"
        PREFIX   -> "Prefix"
        WILDCARD -> "Wildcard"
    }

    val placeholder: String get() = when (this) {
        EXACT    -> "0912345678"
        PREFIX   -> "0900  (blocks 0900xxxxxx)"
        WILDCARD -> "09xx1234  (x = any digit)"
    }

    val hint: String get() = when (this) {
        EXACT    -> "Blocks this exact number."
        PREFIX   -> "Blocks all numbers starting with this prefix."
        WILDCARD -> "Use 'x' as a wildcard for any single digit."
    }
}

File: ios/CallBlocker/Models/PatternType.swift
import Foundation

/// The kind of pattern a BlockRule uses to match phone numbers.
enum PatternType: String, Codable, CaseIterable, Identifiable {
    /// Exact 10-digit number (e.g. "0912345678")
    case exact
    /// Leading digits that must match (e.g. "0900" matches any 0900xxxxxx)
    case prefix
    /// Fixed-length pattern where 'x' matches any single digit (e.g. "09xx1234")
    case wildcard

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .exact:    return "Exact"
        case .prefix:   return "Prefix"
        case .wildcard: return "Wildcard"
        }
    }

    var placeholder: String {
        switch self {
        case .exact:    return "0912345678"
        case .prefix:   return "0900  (blocks 0900xxxxxx)"
        case .wildcard: return "09xx1234  (x = any digit)"
        }
    }
}
